test/libyul/ObjectParser.cpp

```diff
@@ -112,7 +112,8 @@ std::tuple<std::optional<SourceNameMap>, ErrorList> tryGetSourceLocationMapping(
 
 	ErrorList errors;
 	ErrorReporter reporter(errors);
-	Dialect const& dialect = yul::EVMDialect::strictAssemblyForEVM(EVMVersion::berlin());
+	// TODO: Add EOF support
+	Dialect const& dialect = yul::EVMDialect::strictAssemblyForEVM(EVMVersion::berlin(), std::nullopt);
```

Comment:
Not sure why this one is hard-coded to `berlin`, but it should at least be the default  version. I'd change it to that if it won't test fail:

```suggestion
	Dialect const& dialect = yul::EVMDialect::strictAssemblyForEVM(EVMVersion{}, std::nullopt);
```
Ideally we should later change it to properly use the version from `CommonOptions`.

BTW: I'd put any changes related to EVM version at least in a separate commit, since they're really unrelated to introduction of EOF version.